src/cgtop/cgtop.c

```diff
@@ -852,7 +852,7 @@ static int parse_argv(int argc, char *argv[]) {
                         }
 
                         arg_recursive = r;
-                        recursive_unset = r == 0;
+                        arg_recursive_unset = r == 0;
```

Comment:
I just move the code around. Before it was in a single function, so a stack variable was enough, and now it needs to be accessed from two places, so I made it global.

The code differentiates the case when arg_recursive was explicitly set.